src/m365/commands/setup.ts

```diff
@@ -106,59 +205,218 @@ class SetupCommand extends AnonymousCommand {
       message: 'How experienced are you in using the CLI?',
       choices: [
-        { name: 'Beginner', value: 'Beginner' },
-        { name: 'Proficient', value: 'Proficient' }
+        { name: 'Beginner', value: CliExperience.Beginner },
+        { name: 'Proficient', value: CliExperience.Proficient }
       ]
     };
     preferences.experience = await cli.promptForSelection(experienceConfig);
 
     const summaryConfig: ConfirmationConfig = {
-      message: this.getSummaryMessage(this.getSettings(preferences))
+      message: this.getSummaryMessage(preferences)
     };
     preferences.summary = await cli.promptForConfirmation(summaryConfig);
 
-    if (preferences.summary) {
-      // used only for testing. Normally, we'd get the settings from the answers
-      /* c8 ignore next 3 */
-      if (!settings) {
-        settings = this.getSettings(preferences);
-      }
+    if (!preferences.summary) {
+      return;
+    }
 
+    // used only for testing. Normally, we'd get the settings from the answers
+    /* c8 ignore next 3 */
+    if (!settings) {
+      settings = this.getSettings(preferences);
+    }
+
+    await logger.logToStderr('');
+    await logger.logToStderr('Configuring settings...');
+    await logger.logToStderr('');
+
+    await this.configureSettings({ preferences, settings, silent: false, logger });
+
+    if (!this.verbose) {
       await logger.logToStderr('');
-      await logger.logToStderr('Configuring settings...');
-      await logger.logToStderr('');
+      await logger.logToStderr(chalk.green('DONE'));
+    }
+  }
 
-      await this.configureSettings(settings, false, logger);
+  private async configureExistingEntraApp(logger: Logger): Promise<Preferences> {
+    await logger.logToStderr('Please provide the details of the existing app registration.');
+    let clientCertificateFile: string | undefined;
+    let clientCertificateBase64Encoded: string | undefined;
+    let clientCertificatePassword: string | undefined;
 
-      if (!this.verbose) {
-        await logger.logToStderr('');
-        await logger.logToStderr(chalk.green('DONE'));
+    const clientId = await cli.promptForInput({
+      message: 'Client ID:',
+      /* c8 ignore next */
+      validate: value => value.length > 0
+    });
+    const tenantId = await cli.promptForInput({
+      message: 'Tenant ID (leave common if the app is multitenant):',
+      default: 'common'
+    });
+    const clientSecret = await cli.promptForInput({
+      message: 'Client secret (leave empty if you use a certificate or a public client):'
+    });
+    if (!clientSecret) {
+      clientCertificateFile = await cli.promptForInput({
+        message: `Path to the client certificate file (leave empty if you want to specify a base64-encoded certificate string):`
+      });
+      if (!clientCertificateFile) {
+        clientCertificateBase64Encoded = await cli.promptForInput({
+          message: 'Base64-encoded certificate string:'
```

Comment:
shouldn't we specify that you may also leave this empty and it will be perfectly fine? I mean you may just use the `clientId` and `tenantId` and you may still `login`

Reply:
Good point, will update.